collections/_posts/2023-04-03-typelevel_toolkit.md

```diff
@@ -0,0 +1,108 @@
+<br />
+Typelevel Toolkit shines with `scala-cli`, but it can also be used by sbt or mill if that is preferred.
+<br /><br />
+More concretely this means your next ad-hoc script won't be Bash or Python spaghetti, but Scala code that can be a joy to hack on as time goes on, without the boilerplate.
+
+You can use the toolkit by using a
+[scala-cli directive](https://scala-cli.virtuslab.org/docs/guides/using-directives)
+
+```scala
+//> using dep "org.typelevel::toolkit::0.0.4"
+```
+
+This will pull in the `typelevel/toolkit` dependency and then you're just an import away from your first pure functional "Hello, World!":
+
+```scala
+import cats.effect.*
+
+object Hello extends IOApp.Simple {
+  def run = IO.println("Hello, World!")
+}
+```
+
+You can compile and run this program by using a single command: `scala-cli run Main.scala`.
+
+
+A "Hello, World!" is only the start, the goal here is to make functional programming friendly and practical. As such, Typelevel toolkit comes with [examples](https://typelevel.org/toolkit/examples.html) that introduces beginners on how one can use the included libraries to achieve common tasks.
+
+
+For the full list of libraries included in Typelevel Toolkit, please see the overview: [https://typelevel.org/toolkit/#overview](https://typelevel.org/toolkit/#overview). If you feel like anything is missing, [join the discussion](https://github.com/typelevel/toolkit/issues/1).
+
+# We can have nice things
+
+Typelevel libraries are production-proven, well tested, build upon rock solid semantics, and almost all have Scala 3 support.
+However their entry-point is higher than your usual scripting language. Pure functional programming has a reputation of being hard to learn, and Typelevel Toolkit is a way to play in that world, without learning an entire ecosystem first.
+
+The toolkit includes, among others, a [HTTP client](https://http4s.org/), [CSV parsing library](https://fs2-data.gnieh.org/documentation/csv/) and functions for [handling files cross-platform](https://fs2.io/#/io), **with support for other runtimes besides the JVM**. This means that your scripts can run in a [JavaScript environment](https://scala-cli.virtuslab.org/docs/guides/scala-js), thanks to [scala-js](https://www.scala-js.org/). Or you can use [scala-native](https://github.com/scala-native/scala-native) to get a [native binary](https://scala-cli.virtuslab.org/docs/guides/scala-native), just like Rust and Go!
+
+`scala-cli` again, makes things easy for us by having simple commands to compile to a certain target:
+
+```sh
+# To compile to JavaScript
+scala-cli Main.scala --js  
+
+# To target native
+scala-cli Main.scala --native
+```
+
+If you just want to explore Typelevel Toolkit, you can open up a REPL using a file that has the `using dep "org.typelevel::toolkit::0.0.4"` directive and you can use all the libraries in the REPL:
```

Comment:
Making a file isn't even necessary; you can `scala-cli --dep "org.typelevel::toolkit::0.0.4"`

Reply:
TIL that flag works with the repl command! :P